src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -133,24 +133,34 @@ protected override void CheckInterfaces(DiagnosticBag diagnostics)
                 var conversions = new TypeConversions(corLibrary);
                 var location = singleDeclaration.NameLocation;
 
-                foreach (var @interface in interfaces)
+                foreach (var pair in interfaces)
                 {
-                    @interface.CheckAllConstraints(conversions, location, diagnostics);
-                }
+                    MultiDictionary<NamedTypeSymbol, NamedTypeSymbol>.ValueSet set = pair.Value;
 
-                if (interfaces.Count > 1)
-                {
-                    var seenInterfaces = new Dictionary<NamedTypeSymbol, NamedTypeSymbol>(EqualsIgnoringTupleNames);
-                    foreach (var @interface in interfaces)
+                    foreach (var @interface in set)
                     {
-                        NamedTypeSymbol other;
-                        if (seenInterfaces.TryGetValue(@interface, out other))
-                        {
-                            diagnostics.Add(ErrorCode.ERR_DuplicateInterfaceWithTupleNamesInBaseList, location, @interface, other, this);
-                        }
-                        else
+                        @interface.CheckAllConstraints(conversions, location, diagnostics);
+                    }
+
+                    if (set.Count > 1)
+                    {
+                        NamedTypeSymbol other = null;
+                        foreach (var @interface in set)
```

Comment:
> Can we not merge these loops so we can avoid looping through the set twice?

I guess we could, but we had two loops before this change. I do not see much benefit in optimizing for an error case, we are not going to get here otherwise. Let me know if you still want me to refactor the code 

---
In reply to: [245106096](https://github.com/dotnet/roslyn/pull/32041#discussion_r245106096) [](ancestors = 245106096)

Reply:
It's fine, it was just a thought I had as I was reading the method.

---
In reply to: [245111937](https://github.com/dotnet/roslyn/pull/32041#discussion_r245111937) [](ancestors = 245111937,245106096)